### ISSUE TITLE
OneWire support

### ISSUE THREAD

**Issue by David-OConnor** [stm32-rs/stm32f3xx-hal#104]
OneWire is a digital protocol used by a very popular temperature sensor, the DS1820B. I'm unable to get it working with this crate, and am wonder what you think the best approach is.

I'm suspicious [this recent issue](https://github.com/stm32-rs/stm32f3xx-hal/issues/84) is related.

I think the issue is that the GPIO pins this library uses are generally input or output only, and don't implement both. Is this correct? The specific issue I'm running into with [this OneWire crate](https://github.com/kellerkindt/onewire/blob/master/src/lib.rs) is that it expects the pin to implement this:
```rust
pub trait OpenDrainOutput<E>: OutputPin<Error = E> + InputPin<Error = E> {}
```

Is this an issue of implementing both `InputPin` and `OutputPin` for gpio pins configured as open drain? I looked through [this source file](https://github.com/stm32-rs/stm32f3xx-hal/blob/master/src/gpio.rs) to try to find a solution, either robust, or a hack to make OneWire work, but am not sure how to deal with the macros. Is it just an issue of making a way to configure a pin with both the `InputPin` and `OutputPin` methods? Ie `set_low()` and `is_low()`? Ie a copy+paste job.

Thoughts? Is this the same thing as the issue I linked near the top?

**Comment by ra-kete** [comment]
@David-OConnor I believed this was fixed with merging #114. Can you confirm?

**Comment by David-OConnor** [comment]
Affirm. Closing!
